examples/multiple_viewer_widget.py

```diff
@@ -0,0 +1,449 @@
+"""
+This is an example on how to have more than one viewer in the same napari window.
+Additional viewers state will be synchronized with the main viewer.
+Switching to 3D display will only impact the main viewer.
+
+This example also contain option to enable cross that will be moved to the
+current dims point (`viewer.dims.point`).
+"""
+
+from copy import deepcopy
+
+import numpy as np
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QDoubleSpinBox,
+    QPushButton,
+    QSplitter,
+    QTabWidget,
+    QVBoxLayout,
+    QWidget,
+)
+from superqt.utils import qthrottled
+from packaging.version import parse as parse_version
+
+import napari
+from napari.components.layerlist import Extent
+from napari.components.viewer_model import ViewerModel
+from napari.layers import Image, Labels, Layer, Vectors
+from napari.qt import QtViewer
+from napari.utils.action_manager import action_manager
+from napari.utils.events.event import WarningEmitter
+from napari.utils.notifications import show_info
+
+NAPARI_GE_4_16 = parse_version(napari.__version__) > parse_version("0.4.16")
+
+
+def copy_layer_le_4_16(layer: Layer, name: str = ""):
+    res_layer = deepcopy(layer)
+    # this deepcopy is not optimal for labels and images layers
+    if isinstance(layer, (Image, Labels)):
+        res_layer.data = layer.data
+
+    res_layer.metadata["viewer_name"] = name
+
+    res_layer.events.disconnect()
+    res_layer.events.source = res_layer
+    for emitter in res_layer.events.emitters.values():
+        emitter.disconnect()
+        emitter.source = res_layer
+    return res_layer
+
+
+def copy_layer(layer: Layer, name: str = ""):
+    if NAPARI_GE_4_16:
+        return copy_layer_le_4_16(layer, name)
+
+    res_layer = Layer.create(*layer.as_layer_data_tuple())
+    res_layer.metadata["viewer_name"] = name
+    return res_layer
+
+
+def get_property_names(layer: Layer):
+    klass = layer.__class__
+    res = []
+    for event_name, event_emitter in layer.events.emitters.items():
+        if isinstance(event_emitter, WarningEmitter):
+            continue
+        if event_name in ("thumbnail", "name"):
+            continue
+        if (
+            isinstance(getattr(klass, event_name, None), property)
+            and getattr(klass, event_name).fset is not None
+        ):
+            res.append(event_name)
+    return res
+
+
+def center_cross_on_mouse(
+    viewer_model: napari.components.viewer_model.ViewerModel,
+):
+    """move the cross to the mouse position"""
+
+    if not getattr(viewer_model, "_mouse_over_canvas", True):
```

Comment:
Same comment as above — overall, this example would be much cleaner if we didn't cater for old versions. Since the purpose of examples is educational, I think clean code is *much* more important than maximum compatibility. I would remove this.